pkg/api/validation/schema.go

```diff
@@ -88,6 +101,29 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
 	if kind == nil {
 		return fmt.Errorf("kind not set")
 	}
+	if kind.(string) == "List" {
+		items, err := unpackObject(obj.(map[string]interface{}))
+		if err != nil {
+			return err
+		}
+		for _, item := range items {
+			fields := item.(map[string]interface{})
+			groupVersion := fields["apiVersion"]
+			if groupVersion == nil {
+				return fmt.Errorf("list item apiVersion not set")
+			}
+			kind := fields["kind"]
```

Comment:
I'm against name shadowing.  Please choose non-conflicting names for `groupVersion` and `kind` in this block.